lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,457 @@
+            # conditions for adding a key not fulfilled
+            return False
+
+        if self._module.params['state'] == 'absent':
+            self._module.fail_json(msg="Contradiction in setup: Asking to "
+                                   "add a key to absent LUKS.")
+
+        return True
+
+    def luks_remove_key(self):
+        if (self._module.params['device'] is None or
+                self._module.params['remove_key'] is None):
+            # conditions for removing a key not fulfilled
+            return False
+
+        if self._module.params['state'] == 'absent':
+            self._module.fail_json(msg="Contradiction in setup: Asking to "
+                                   "remove a key from absent LUKS.")
+
+        return True
+
+    def luks_remove(self):
+        return (self._module.params['device'] is not None and
+                self._module.params['state'] == 'absent' and
+                self.is_luks(self._module.params['device']))
+
+
+def run_module():
+    # available arguments/parameters that a user can pass
+    module_args = dict(
+        state=dict(type='str', choices=['present', 'absent'], required=False, default='present'),
+        device=dict(type='str', required=False),
+        open=dict(type='bool', required=False, default=False),
+        name=dict(type='str', required=False),
+        key=dict(type='path', required=False),
+        new_key=dict(type='path', required=False),
+        remove_key=dict(type='path', required=False)
+    )
+
+    # seed the result dict in the object
+    result = dict(
+        changed=False,
+        name=None
+    )
+
+    module = AnsibleModule(argument_spec=module_args,
+                           supports_check_mode=False)
+
+    crypt = CryptHandler(module)
+    conditions = ConditionsHandler(module)
+
+    # The conditions are in order to allow more operations in one run.
+    # (e.g. create luks and add a key to it)
+
+    # luks create
+    if conditions.luks_create():
+        try:
+            crypt.run_luks_create(module.params['device'],
+                                  module.params['key'])
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['changed'] = True
+
+    # luks open
+    if conditions.luks_open():
+        name = module.params['name']
+        if name is None:
+            try:
+                name = crypt.generate_luks_name(module.params['device'])
+            except ValueError as e:
+                module.fail_json(msg="luks_device error: %s" % e)
+        try:
+            crypt.run_luks_open(module.params['device'],
+                                module.params['key'],
+                                name)
+        except ValueError as e:
+            module.fail_json(msg="luks_device error: %s" % e)
+        result['name'] = name
+        result['changed'] = True
+
+    # luks close
+    if conditions.luks_close():
+        if module.params['device'] is not None:
```

Comment:
Why do you have no `try/except` block around these calls?

Reply:
`conditions.luks_close` does not raise any exception to be caught.

Reply:
The line below calls `get_container_name_by_device`, which can raise `ValueError` according to its documentation.